tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -80,58 +81,63 @@ protected:
    class RFieldContext {
       friend class RNTupleProcessor;
       friend class RNTupleChainProcessor;
+      friend class RNTupleJoinProcessor;
 
    private:
       std::unique_ptr<RFieldBase> fProtoField;
       std::unique_ptr<RFieldBase> fConcreteField;
       REntry::RFieldToken fToken;
+      std::string fNTupleName;
+      bool fIsAuxiliary;
 
    public:
-      RFieldContext(std::unique_ptr<RFieldBase> protoField, REntry::RFieldToken token)
-         : fProtoField(std::move(protoField)), fToken(token)
+      RFieldContext(std::unique_ptr<RFieldBase> protoField, REntry::RFieldToken token, std::string_view ntupleName,
+                    bool isAuxiliary = false)
+         : fProtoField(std::move(protoField)), fToken(token), fNTupleName(ntupleName), fIsAuxiliary(isAuxiliary)
       {
       }
 
       const RFieldBase &GetProtoField() const { return *fProtoField; }
-      /// We need to disconnect the concrete fields before swapping the page sources
+      /// Concrete pages need to be reset explicitly before the page source they belong to is destroyed.
       void ResetConcreteField() { fConcreteField.reset(); }
       void SetConcreteField() { fConcreteField = fProtoField->Clone(fProtoField->GetFieldName()); }
+      bool IsAuxiliary() const { return fIsAuxiliary; }
+      const std::string &GetNTupleName() const { return fNTupleName; }
+      std::string GetQualifiedFieldName() const
+      {
+         if (fIsAuxiliary)
+            return fNTupleName + "." + fProtoField->GetFieldName();
```

Comment:
Should we maybe have this as a generic `fPrefix`, taking away information about `fIsAuxiliary` and `fNTupleName` concatenated with a dot?